### ISSUE TITLE
Circular definitions in FIBO Prod

### ISSUE THREAD

**Issue by mereolog** [edmcouncil/fibo#1050]
ONTOLOGY_GUIDE.md says that "Definitions must not be circular, i.e., the class, property, or individual name must not be used in the definition itself."

I take it that it implies that 
if:
label of A1 is used in definition of A2
label of A2 is used in definition of A3
etc.
label of An is used in definition of A1
then:
there is a circular chain of definitions for A1, A2, ..., An because A1 is effectively defined by means of A1 (and some for other As).

This interpretation assumes that the aforementioned convention speaks about labels and not about parts of URIs (local fragments thereof). The assumption can be rationalised as follows:
- definitions are in plain English, so search for possible circularities should focus on labels, which are also in English
- definienda should be replaceable with definitions in sentences, which, I presume, usually do not contain URIs or their parts.
Still some of the cases collected in the attachment are valid also for local fragments.

I think that there is quite a lot of such chains in the current FIBO Prod - see [fibo_circular_definitions_in_fibo_prod_2020_q1.xlsx](https://github.com/edmcouncil/fibo/files/4837779/fibo_circular_definitions_in_fibo_prod_2020_q1.xlsx).

In fact there are two types of such circular definitions.
A. degenerate definitional chains, where label of A1 is used in definition of A1
Take https://spec.edmcouncil.org/fibo/ontology/FND/Law/Jurisdiction/CivilLawJurisdiction as an example, where label is 'civil law jurisdiction' and definition reads 'a civil law jurisdiction' - this indicates an obvious case of missing definition.
The attachment reports more than 800 of such cases.
Most of them concern business center individuals or interest rates and look rather innocent when compared to other cases.
But if we have governance over their labels, perhaps we should rephrase them, e.g., have 'AED-EBOR-Reuters rate benchmark' as a label for https://spec.edmcouncil.org/fibo/ontology/IND/InterestRates/CommonInterestRates/AED-EBOR-Reuters.
B. proper definitional chains.
Consider these two classes:
class_1 > https://spec.edmcouncil.org/fibo/ontology/FND/Places/VirtualPlaces/VirtualLocation
label: 'virtual location'
definition: 'place that is not located in any single physical location; a **network location** without geographic boundaries'
class_2 > https://spec.edmcouncil.org/fibo/ontology/FND/Places/VirtualPlaces/NetworkLocation
label: 'network location'
definition: 'a **virtual location** that may be identified by a network address (an identifier for a node or interface)**'.
Now, 
- definition for class_1 contains label for class_2
- definition for class_2 contains label for class_1
So class_1 is defined by means of class_2, which is defined by means of class_1.
The attachment reports 21 such cases.

Note that some reported cases indicate issues of other sort, e.g., https://spec.edmcouncil.org/fibo/ontology/BE/OwnershipAndControl/Executives/Bylaw is reported because its definition does not follow ISO 704.

**DISCLAIMER: The reported results were obtained by strict search for labels in definitions, so:
- they may contain some false positives, e.g., perhaps case of https://spec.edmcouncil.org/fibo/ontology/SEC/Debt/Bonds/isMandatory is an example of this;
- there might be some false negatives;
due to the grammatical variations of words and phrases.**

**Comment by rivettp** [comment]
Great analysis @mereolog - we should aim to incorporate what you ran as a hygiene test!
I agree we should address both types of circularity.

For other reasons I think we should restrict skos:definition to real (T-box) ontology elements and not use it on (A-Box) Individuals -- for which we should stick to labels and notes/comments. The bulk of these Individuals are either generated (e.g. the ones you mention like interest rates and business centers), or in their own xIndividuals ontologies, so it would be easy to change - though not pressing.
BTW I don't include individuals that are enumerated members of a Class extension - they are really T-box.  

Reasons include that skos:definition is itself defined as "A statement or formal explanation of the meaning of a concept." and that's not what we're providing for A-Box Individuals.
More importantly, if we're trying to generate a dictionary/glossary of terms and definitions, we don't want it cluttered up with these individuals.

**Comment by ElisaKendall** [comment]
This is really helpful - we can work on these as time permits, prioritizing those that are class definitions.  Many of these are skos:definitions on reference data, which I prefer to retain but not test for circularity.  Granted they are generated for the most part, so another approach would be to use something like dct:description, since it is intended to be exactly that rather than a definition per se.  

We have another rule that says we won't use rdfs:comment because it is too general, which I would prefer we continue to follow.
